src/transformers/generation_utils.py

```diff
@@ -2372,19 +2495,22 @@ def group_beam_search(
                         else (outputs.hidden_states,)
                     )
 
+            input_ids = torch.cat([input_ids, current_tokens.unsqueeze(-1)], dim=-1)
+
             model_kwargs = self._update_model_kwargs_for_generation(
                 outputs, model_kwargs, is_encoder_decoder=self.config.is_encoder_decoder
             )
             if model_kwargs["past"] is not None:
                 model_kwargs["past"] = self._reorder_cache(model_kwargs["past"], reordering_indices)
 
-            input_ids = torch.cat([input_ids, current_tokens.unsqueeze(-1)], dim=-1)
```

Comment:
There are 5 different versions of search functions and what I did is to bring them in sync on the order of how the last 4-5 chunks of code is done. e.g. `cur_len` was incremented inconsistently in several places - the only reason it wasn't breaking things is that the functions called using `cur_len` weren't impacting the outcome outside the loop. But it could.

So I thought that while I'm at it, I will use the `greedy_search` as the model and sync the other 4 to have the same order. That is:

````
             # add token and increase length by one
            input_ids = torch.cat([input_ids, next_tokens[:, None]], dim=-1)

            # update sequence length
            if eos_token_id is not None:
                sequence_lengths, unfinished_sequences = self._update_seq_length_for_generation(
                    sequence_lengths, unfinished_sequences, cur_len, next_tokens == eos_token_id
                )

            # update model kwargs
            model_kwargs = self._update_model_kwargs_for_generation(
                outputs, model_kwargs, is_encoder_decoder=self.config.is_encoder_decoder
            )

            # increase cur_len
            cur_len = cur_len + 1

            # stop when there is a </s> in each sentence, or if we exceed the maximum length
            if unfinished_sequences.max() == 0 or stopping_criteria(input_ids, scores):
                if not synced_gpus:
                    break
                else:
                    this_peer_finished = True
```